server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V4349OrganizationSecretMigrate.java

```
@@ -0,0 +1,82 @@
+
+    private JdbcTemplate jdbcTemplate;
+    private TransactionTemplate transactionTemplate;
+
+    @Override
+    public void migrate(DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.transactionTemplate = new TransactionTemplate(new DataSourceTransactionManager(dataSource));
+        List<OrganizationEntity> organizationList = listOrganizations();
+        if (CollectionUtils.isEmpty(organizationList)) {
+            log.info("organization secret migrate skipped, organization list is empty");
+            return;
+        }
+        log.info("organization secret migrate started, organizationCount={}", organizationList.size());
+        int total = 0;
+        for (OrganizationEntity organization : organizationList) {
+            organization.setSecret(EncodeUtils.base64EncodeToString(organization.getSecret().getBytes()));
+            total += migrateForOrganization(organization);
+        }
+        log.info("organization secret migrated, organizationCount={}, total={}", organizationList.size(), total);
+    }
+
+    private List<OrganizationEntity> listOrganizations() {
+        String sql = "select `id`, `secret` from iam_organization";
+        return jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(OrganizationEntity.class));
+    }
+
+    private int migrateForOrganization(OrganizationEntity organization) {
+        String sql = "update iam_organization set secret=? where id=?";
+        AtomicInteger row = new AtomicInteger();
+        transactionTemplate.execute(status -> {
+            try {
+                row.set(jdbcTemplate.update(sql, organization.getSecret(),
+                        organization.getId()));
+            } catch (Exception e) {
+                log.error("organization secret migrate failed, organizationId={}, error={}", organization.getId(),
+                        e.getMessage());
+                status.setRollbackOnly();
```

1. What will happen if you swallow the exception?
2. What if just a part of the data has been migrated because of an exception? Is your migration repeatable?

---

1. throw a RuntimeException.
2. Use batch updates and ensure that only both success and failure are allowed.